pkg/tfgen/generate_nodejs.go

```diff
@@ -479,6 +479,29 @@ func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (stri
 		w.Writefmtln("")
 	}
 
+	w.Writefmtln("    private static readonly __pulumiType = '%s';", res.info.Tok)
+
+	if !res.IsProvider() {
+		w.Writefmtln("")
+		w.Writefmtln("    /**")
+		w.Writefmtln("     * Returns true if the given object is an instance of %s.  This is designed to work even", name)
+		w.Writefmtln("     * when multiple copies of the Pulumi SDK have been loaded into the same process.")
+		w.Writefmtln("     */")
+		w.Writefmtln("    public static isInstance(obj: any): obj is %s {", name)
+		w.Writefmtln("        if (obj === undefined || obj === null) {")
+		w.Writefmtln("            return false;")
+		w.Writefmtln("        }")
+		w.Writefmtln("")
+		w.Writefmtln("        const t = obj['__pulumiType'];")
+		w.Writefmtln("        if (typeof t !== 'string') {")
+		w.Writefmtln("            return false;")
+		w.Writefmtln("        }")
```

Comment:
This `typeof` test seems unnecessary - it cannot be `===` below` if it's not the same type.

Reply:
👍

Reply:
This is basically copied from the pulumi/pulumi implementation of `isInstance`

Reply:
the existing isInstance does things differently (and can't change because it literally needs to always operate the same in SxS scenarios), so you def don't have to follow it :)

Looking at it, i can see i was too paranoid.  It doesn't need the `typeof` check.  The `===` check is sufficient against that precise value that it's looking for.